packages/lib/src/vis/shared/AxisSystemProvider.tsx

```diff
@@ -45,8 +47,11 @@ function AxisSystemProvider(props: PropsWithChildren<Props>) {
     children,
   } = props;
 
-  const availableSize = useThree((state) => state.size);
-  const visSize = getSizeToFit(availableSize, visRatio);
+  const canvasSize = useThree((state) => state.size);
```

Comment:
Does this change anything with regards to rerenders ?

I mean, now that `canvasSize` is in the `context`, it means a change in `size` will change the context and therefore all the consumers, right? But I don't think this is different from the previous behaviour since `visSize` was depending on `canvasSize` and in the context.

Reply:
Precisely, no change to the number of re-renders, since:

1.  `AxisSystemProvider` used to already re-render on canvas `size` changes;
2. It doesn't retrieve anything else from `useThree` and doesn't have any `useState` that could trigger extra renders.